plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -225,6 +225,11 @@ protected static QueryRunner createHiveQueryRunner(HiveQueryRunner.Builder<?> bu
                 // Make weighted split scheduling more conservative to avoid OOMs in test
                 .addHiveProperty("hive.minimum-assigned-split-weight", "0.5")
                 .addHiveProperty("hive.partition-projection-enabled", "true")
+                // This is needed for Ion native trino support as the implementation is in progress and can be
+                // backwards incompatible. It requires explicitly opting to use this feature.
+                // TODO: In future this property should change to `true` as default and then the following statement can
+                //  be removed.
+                .addHiveProperty("hive.ion.nativetrino", "true")
```

Comment:
Is the apparent low test coverage for the new functionality intentional?

Reply:
I don't believe the test coverage for the new functionality actually is low.

We have added the `TestIonFormat` in `lib/trino-hive-formats` and the `IonPageSourceSmokeTest` as well the addition of `testIonWithBinaryEncoding` to `TestHiveFileFormats`. And that flag is enabled for them. I see that those tests do run when I build locally.

Can you point me to the artifacts that indicate there is low test coverage?

Reply:
I did check that the `TestHiveConnectorTest` runs against the `ION` format and checked the coverage for our tests. There are a few gaps that we can and will address, but on the whole the coverage seems pretty robust.